File: src/module/clean_fields.js
const fs = require('fs');
const path = require('path');

const vueDir = path.join('d:', 'develop_python', 'django', 'autrom_vue', 'src', 'module');

function walk(dir) {
    let results = [];
    const list = fs.readdirSync(dir);
    list.forEach(file => {
        file = path.join(dir, file);
        const stat = fs.statSync(file);
        if (stat && stat.isDirectory()) {
            results = results.concat(walk(file));
        } else if (file.endsWith('Register.vue') || file.endsWith('Update.vue')) {
            results.push(file);
        }
    });
    return results;
}

const files = walk(vueDir);
files.forEach(file => {
    let content = fs.readFileSync(file, 'utf8');
    const orig = content;
    
    // Remove UIInputText/UIAutocomplete for com and _create_at and _update_at
    const removeRegex = /<div>\s*<(UIInputText|UIAutocomplete|UIInputDate)[^>]*name=\"(com|[a-zA-Z0-9_]+_(create_at|update_at))\"[\s\S]*?<\/\1>\s*<\/div>|<div>\s*<(UIInputText|UIAutocomplete|UIInputDate)[^>]*name=\"(com|[a-zA-Z0-9_]+_(create_at|update_at))\"[^>]*\/>\s*<\/div>/g;
    content = content.replace(removeRegex, '');

    // In Register.vue, remove sub completely
    if (file.endsWith('Register.vue')) {
        const removeSubRegex = /<div>\s*<(UIInputText|UIAutocomplete|UIInputDate)[^>]*name=\"sub\"[\s\S]*?<\/\1>\s*<\/div>|<div>\s*<(UIInputText|UIAutocomplete|UIInputDate)[^>]*name=\"sub\"[^>]*\/>\s*<\/div>/g;
        content = content.replace(removeSubRegex, '');
    }

    if (content !== orig) {
        fs.writeFileSync(file, content);
        console.log('Updated ' + file);
    }
});

File: src/module/clean_fields.py
import os
import re

vue_dir = r'd:\develop_python\django\autrom_vue\src\module'

def remove_blocks(content, names):
    for name in names:
        pattern = r'\s*<div>\s*<(?:UIInputText|UIAutocomplete|UIInputDate)[^>]*name=[\"\']' + re.escape(name) + r'[\"\'][^>]*>(?:.*?</(?:UIInputText|UIAutocomplete|UIInputDate)>)?\s*</div>'
        content = re.sub(pattern, '', content, flags=re.DOTALL)
    return content

for root, dirs, files in os.walk(vue_dir):
    for file in files:
        if file.endswith('.vue') and (file == 'Register.vue' or file == 'Update.vue'):
            path = os.path.join(root, file)
            with open(path, 'r', encoding='utf-8') as f:
                content = f.read()
            
            names_to_remove = ['com']
            matches = re.findall(r'name=[\"\']([a-zA-Z0-9_]+_(?:create_at|update_at))[\"\']', content)
            names_to_remove.extend(matches)
            
            if file == 'Register.vue':
                names_to_remove.append('sub')
            
            orig_content = content
            content = remove_blocks(content, names_to_remove)
            
            if content != orig_content:
                with open(path, 'w', encoding='utf-8') as f:
                    f.write(content)
                print(f'Updated {path}')
